src/DistributedComputations/distributed_architectures.jl

```diff
@@ -51,10 +25,47 @@ Return `Partition` representing the division of a domain into
 `Rx` parts in `x` and `Ry` parts in `y` and `Rz` parts in `z`,
 where `x, y, z` are the first, second, and third dimension
 respectively.
+
+`Rx`, `Ry` and `Rz` can be `Int` or `Equal`, `Fractional` or `Sizes` 
+(see below)
 """
-Partition(Rx::Number, Ry::Number=1, Rz::Number=1) = Partition{Rx, Ry, Rz}()
-Partition(lengths::Array{Int, 1}) = Partition(reshape(lengths, length(lengths), 1))
-Base.size(::Partition{<:Any, Rx, Ry, Rz}) where {Rx, Ry, Rz} = (Rx, Ry, Rz)
+Partition(Rx)     = Partition(validate_partition(Rx, 1, 1)...)
+Partition(Rx, Ry) = Partition(validate_partition(Rx, Ry, 1)...)
+
+function Partition(; x = 1, y = 1, z = 1)
```

Comment:
I think we need a docstring with a few examples of uniform, non-uniform, etc partitioning?